### PR TITLE
3D fixes

### PR DESCRIPTION
**Pull request by Bod9001**
Fixes codes Scan tests

### Changelog:
CL: [Improvement] adds in the framework for requesting users consent for microphone access
CL: [Fix] fixes up 3D mode

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		{
+			return Microphone.IsRecording(deviceName);
+		}
+
+		public static int GetPosition(string deviceName)
+		{
+			if (MicEnabled == false)
+			{
+				//TODO TEMP
+				//return 0;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs:34]: nani

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs:34]: wouldn't it be better to do this check wherever this and the other method are called?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs:34]: oh? nah That would be on the unsecured code side?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs:34]: uhmm, ok. How do you intend to solve this TODO?

File: UnityProject/Assets/Scripts/Managers/Manager3D.cs [hunk 2]
```diff
@@ -50,11 +50,11 @@ public void PromptConvertTo3D()
 		DynamicChoiceUI.ClientDisplayChoicesNotNetworked("Doom? **Hold !!!TAB to use mouse, will freeze for a few seconds", " would you like to activate DOOM mode?, is a WIP so is buggy will freeze for a few seconds *Hold !TAB! to use mouse!*, Would you also like music to accompany it? ",
 			new List<DynamicUIChoiceEntryData>()
 			{
-				new DynamicUIChoiceEntryData()
-				{
-					ChoiceAction =  Ceilings,
-					Text = " Give me DOOM with ceilings! ( Performance heavy )"
-				},
+				// new DynamicUIChoiceEntryData()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Managers/Manager3D.cs:53]: why is this commented out?

File: UnityProject/Assets/Scripts/UI/Systems/IngameMenu/GUI_IngameMenu.cs [hunk 3]
```diff
@@ -90,8 +90,8 @@ public void OpenMenuPanel()
 			_ = SoundManager.Play(CommonSounds.Instance.Click01);
 			Loggy.Log($"Opening {menuWindow.name} menu", Category.UI);
 			menuWindow.SetActive(true);
-			InfoPanelMessageClient.Send();
-			if (UIManager.Instance.ServerInfoPanelWindow != null) UIManager.Instance.ServerInfoPanelWindow.SetActive(true);
+			//InfoPanelMessageClient.Send();
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/IngameMenu/GUI_IngameMenu.cs:93]: why is this commented out?

Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Systems/IngameMenu/GUI_IngameMenu.cs:93]: don't comment out code, remove it

File: UnityProject/Assets/Scripts/Managers/Manager3D.cs [hunk 4]
```diff
@@ -50,11 +50,11 @@ public void PromptConvertTo3D()
 		DynamicChoiceUI.ClientDisplayChoicesNotNetworked("Doom? **Hold !!!TAB to use mouse, will freeze for a few seconds", " would you like to activate DOOM mode?, is a WIP so is buggy will freeze for a few seconds *Hold !TAB! to use mouse!*, Would you also like music to accompany it? ",
 			new List<DynamicUIChoiceEntryData>()
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: why is this removed?

File: UnityProject/Assets/Scripts/UniMic/Runtime/Mic.cs [hunk 5]
```diff
@@ -3,6 +3,9 @@
 using System.Collections;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UniMic/Runtime/Mic.cs:3]: if the code for UniMic is in the codebase, what stops forks from modifying this very code and bypass the secure stuff code?

File: UnityProject/Assets/Scripts/Managers/Manager3D.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:53]: was buggy

File: UnityProject/Assets/Scripts/UI/Systems/IngameMenu/GUI_IngameMenu.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Systems/IngameMenu/GUI_IngameMenu.cs:93]: It was annoying since it would show multiple things when pressing escape and then you had to exit out of them individually

File: UnityProject/Assets/Scripts/Managers/Manager3D.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: is buggy

File: UnityProject/Assets/Scripts/UniMic/Runtime/Mic.cs (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UniMic/Runtime/Mic.cs:3]: well nothing can access the microphone, it will go through secure stuff to access the microphone

File: UnityProject/Assets/Scripts/Managers/Manager3D.cs (hunk 4 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: how is it buggy?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: ceiling is offset weirdly and then the floor doesnn't get set properly

File: UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/SecureStuff/MicrophoneAccess.cs:34]: when hub is pushed out Comment the code

File: UnityProject/Assets/Scripts/Managers/Manager3D.cs (hunk 4 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: ...then why remove all options?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: I haven't have only removed one xD

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/Manager3D.cs:51]: okay github is being weird on my end then, had to open the entire file to see that it is the only one commented out
